### PR TITLE
Config logic

### PR DESCRIPTION
**Pull request by youngsterxyf**
try to fix issue #1597 issue #1601 

Make a little adjustment to the logic of config module：

- Add `beego.LoadAppConfig`：it will parse config file immediately when called
- No more to automatically parse config file in `initBeforeHTTPRun`
- `beego.LoadAppConfig` will call `parseConfig`, the last part of `parseConfig` will try to init log writers again

### CODE REVIEW

**Comment by youngsterxyf** [comment]
@astaxie @JessonChan

File: staticfile_test.go [hunk 1]
```diff
@@ -8,9 +8,11 @@ import (
 	"io/ioutil"
 	"os"
 	"testing"
+	"path/filepath"
 )
 
-const licenseFile = "./LICENSE"
+var currentWorkDir, _ = os.Getwd()
+var licenseFile = filepath.Join(currentWorkDir, "LICENSE")
```
Comment (astaxie) [inline staticfile_test.go]: why not just use workPath

File: config.go [hunk 2]
```diff
@@ -286,9 +286,52 @@ func ParseConfig() (err error) {
...
+			fmt.Printf("%s with the config `%s` got err:%s\n", adaptor, config, err)
+		}
+	}
+	SetLogFuncCall(BConfig.Log.FileLineNum)
+
 	return nil
 }
 
+// LoadAppConfig allow developer to apply a config file
+func LoadAppConfig(configPath string, adapterName string) error {
```
Comment (astaxie) [inline config.go]: Could you change the sequence adapterName and configPath?

File: config.go [hunk 3]
```diff
@@ -181,26 +186,19 @@ func init() {
...
-// now only support ini, next will support json.
-func ParseConfig() (err error) {
-	if AppConfigPath == "" {
-		// initialize default configurations
-		AppConfigPath = filepath.Join(AppPath, "conf", "app.conf")
-		if !utils.FileExists(AppConfigPath) {
-			AppConfig = &beegoAppConfig{config.NewFakeConfig()}
-			return
-		}
+	AppConfigPath = filepath.Join(AppPath, "conf", "app.conf")
```
Comment (astaxie) [inline config.go]: if change to this style, why we still need AppConfigPath as public ?

File: staticfile_test.go (hunk 1 above)
Comment (youngsterxyf) [inline staticfile_test.go]: using workPath, the test case will fail

File: config.go (hunk 3 above)
Comment (youngsterxyf) [inline config.go]: Yes, not need, but the next release will be a minor version，so i think not changing the Public variable may be better

File: config.go (hunk 2 above)
Comment (youngsterxyf) [inline config.go]: OK.

File: config.go (hunk 3 above)
Comment (astaxie) [inline config.go]: it doesn't make sense. As if this variable still as public. but beego will never use it. As we overwrite it here. And I think we need to just build failed. and let the developer call LoadAppConfig

Comment (youngsterxyf) [inline config.go]: OK.

**Pull request merged**